pdata/plog/json.go

```diff
@@ -54,14 +56,135 @@ func NewJSONUnmarshaler() Unmarshaler {
 }
 
 func newJSONUnmarshaler() *jsonUnmarshaler {
-	return &jsonUnmarshaler{delegate: jsonpb.Unmarshaler{}}
+	return &jsonUnmarshaler{}
 }
 
 func (d *jsonUnmarshaler) UnmarshalLogs(buf []byte) (Logs, error) {
-	ld := otlplogs.LogsData{}
-	if err := d.delegate.Unmarshal(bytes.NewReader(buf), &ld); err != nil {
-		return Logs{}, err
+	iter := jsoniter.ConfigFastest.BorrowIterator(buf)
+	defer jsoniter.ConfigFastest.ReturnIterator(iter)
+	ld := d.readLogsData(iter)
+	if iter.Error != nil {
+		return Logs{}, iter.Error
 	}
 	otlp.MigrateLogs(ld.ResourceLogs)
 	return Logs(internal.LogsFromProto(ld)), nil
 }
+
+func (d *jsonUnmarshaler) readLogsData(iter *jsoniter.Iterator) otlplogs.LogsData {
+	ld := otlplogs.LogsData{}
+	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+		switch f {
+		case "resource_logs", "resourceLogs":
+			iter.ReadArrayCB(func(iterator *jsoniter.Iterator) bool {
+				ld.ResourceLogs = append(ld.ResourceLogs, d.readResourceLogs(iter))
+				return true
+			})
+		default:
+			iter.Skip()
+		}
+		return true
+	})
+	return ld
+}
+
+func (d *jsonUnmarshaler) readResourceLogs(iter *jsoniter.Iterator) *otlplogs.ResourceLogs {
+	rs := &otlplogs.ResourceLogs{}
+	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+		switch f {
+		case "resource":
+			iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+				switch f {
+				case "attributes":
+					iter.ReadArrayCB(func(iter *jsoniter.Iterator) bool {
+						rs.Resource.Attributes = append(rs.Resource.Attributes, json.ReadAttribute(iter))
+						return true
+					})
+				case "droppedAttributesCount", "dropped_attributes_count":
+					rs.Resource.DroppedAttributesCount = json.ReadUint32(iter)
+				default:
+					iter.Skip()
+				}
+				return true
+			})
```

Comment:
This is duplicate with traces and metrics. we should create a ReadResource similar to ReadScope, and do that in a separate PR please.

Reply:
@bogdandrutu  I fix it. https://github.com/open-telemetry/opentelemetry-collector/pull/6023